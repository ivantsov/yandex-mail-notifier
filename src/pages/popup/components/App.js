import PropTypes from 'prop-types';
import React, {Component} from 'react';
import {connect} from 'react-redux';
import * as actions from '../actions';
import UnavailableMessage from './UnavailableMessage/UnavailableMessage';
import Donation from './Donation/Donation';
import Header from './Header/Header';
import List from './List/List';

class App extends Component {
    static propTypes = {
        user: PropTypes.shape({
            authorized: PropTypes.bool.isRequired,
            email: PropTypes.string.isRequired,
        }).isRequired,
        messages: PropTypes.shape({
            unreadCount: PropTypes.number.isRequired,
            items: PropTypes.arrayOf(PropTypes.object).isRequired,
            loading: PropTypes.bool.isRequired,
            error: PropTypes.bool.isRequired,
        }).isRequired,
        settings: PropTypes.shape({
            preferredDomain: PropTypes.string.isRequired,
        }).isRequired,
        loadMessages: PropTypes.func.isRequired,
        updateMessage: PropTypes.func.isRequired,
        openLink: PropTypes.func.isRequired,
        openSettings: PropTypes.func.isRequired,
        reloadApp: PropTypes.func.isRequired,
        openDonationLink: PropTypes.func.isRequired,
    };

    componentDidMount() {
        const {
            user,
            loadMessages,
        } = this.props;

        if (user.authorized) {
            loadMessages();
        }
    }

    render() {
        const {
            user,
            messages: {
                unreadCount,
                items,
                loading,
                error,
            },
            settings: {
                preferredDomain,
            },
            loadMessages,
            updateMessage,
            openLink,
            openSettings,
            reloadApp,
            openDonationLink,
        } = this.props;

        if (!user.authorized) {
            return (
                <UnavailableMessage
                    domain={preferredDomain}
                    reloadApp={reloadApp}
                />
            );
        }

        return (
            <div>
                <Header
                    user={user.email}
                    unreadMessagesCount={unreadCount}
                    disabled={loading}
                    reloadMessages={loadMessages}
                    openLink={openLink}
                    openSettings={openSettings}
                />
                <List
                    loading={loading}
                    error={error}
                    unreadMessagesCount={unreadCount}
                    items={items}
                    onActionClick={updateMessage}
                    openMessage={openLink}
                />
                <Donation onClick={openDonationLink}/>
            </div>
        );
    }
}

export default connect(state => state, actions)(App);
export {App as AppComponent}; // for test purpose
