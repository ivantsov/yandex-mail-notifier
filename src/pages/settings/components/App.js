import PropTypes from 'prop-types';
import React from 'react';
import {connect} from 'react-redux';
import * as actions from '../actions';
import config from './config';
import Field from './Field/Field';

const App = ({settings, updateSettings}) => {
  const fields = config.map((item, index) => (
    <Field
      key={index}
      {...item}
      value={settings[item.name]}
      onChange={updateSettings}
    />
  ));

  return (
    <table>
      <tbody>{fields}</tbody>
    </table>
  );
};

App.propTypes = {
  settings: PropTypes.object.isRequired,
  updateSettings: PropTypes.func.isRequired,
};

export default connect(state => state, actions)(App);
export {App as AppComponent}; // for test purpose
