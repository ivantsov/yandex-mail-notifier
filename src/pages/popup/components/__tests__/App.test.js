import React from 'react';
import renderer from 'react-test-renderer';
import {AppComponent} from '../App';

jest.mock('shared/utils/i18n');
jest.mock('../Header/Header', () => () => <div>Header</div>);
jest.mock('../List/List', () => () => <div>List</div>);
jest.mock('../Donation/Donation', () => () => <div>Donation</div>);

const baseProps = {
  user: {
    authorized: true,
    email: 'test@ya.ru',
  },
  messages: {
    unreadCount: 5,
    items: [{id: 1}, {id: 2}],
    loading: false,
    error: false,
  },
  settings: {
    preferredDomain: 'ru',
  },
  loadMessages: jest.fn(),
  updateMessage: jest.fn(),
  openLink: jest.fn(),
  openSettings: jest.fn(),
  reloadApp: jest.fn(),
  openDonationLink: jest.fn(),
};

function render(props) {
  const component = renderer.create(<AppComponent {...baseProps} {...props} />);

  return component.toJSON();
}

describe('popup/App', () => {
  it('authorized', () => {
    const tree = render();

    expect(tree).toMatchSnapshot();
    expect(baseProps.loadMessages).toHaveBeenCalledTimes(1);
  });

  it('not authorized', () => {
    const tree = render({
      user: {
        ...baseProps.user,
        authorized: false,
      },
    });

    expect(tree).toMatchSnapshot();
    expect(baseProps.loadMessages).not.toBeCalled();
  });
});
