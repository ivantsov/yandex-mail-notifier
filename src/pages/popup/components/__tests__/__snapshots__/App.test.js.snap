// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`popup/App authorized 1`] = `
<div>
  <div>
    Header
  </div>
  <div>
    List
  </div>
  <div>
    Donation
  </div>
</div>
`;

exports[`popup/App not authorized 1`] = `
<div
  className="container"
>
  <div>
    <h1
      className="title"
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "popup.unavailable.title",
          }
        }
      />
    </h1>
    <p
      className="subTitle"
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "popup.unavailable.subTitle",
          }
        }
      />
    </p>
    <ol>
      <li>
        <strong>
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "popup.unavailable.notAuth.label",
              }
            }
          />
        </strong>
        <ul>
          <li>
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "popup.unavailable.notAuth.checkAuth",
                }
              }
            />
          </li>
          <li>
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "popup.unavailable.notAuth.relogin",
                }
              }
            />
          </li>
        </ul>
      </li>
      <li>
        <strong>
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "popup.unavailable.noConnection.label",
              }
            }
          />
        </strong>
      </li>
    </ol>
  </div>
  <p
    className="btnTitle"
  >
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "popup.unavailable.nothingHelped",
        }
      }
    />
  </p>
  <div
    className="btnContainer"
  >
    <button
      className="primary"
      disabled={false}
      onClick={[Function]}
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "popup.unavailable.reloadBtn",
          }
        }
      />
    </button>
  </div>
</div>
`;
